### PR TITLE
Selection works with PeriodIndex-like indexes 

### PR DESCRIPTION
**Pull request by maximilianr**
Resolves the most pressing issue in #645

### CODE REVIEW

File: xray/core/indexing.py [hunk 1]
```diff
@@ -368,7 +368,12 @@ class PandasIndexAdapter(utils.NDArrayMixin):
     def __init__(self, array, dtype=None):
         self.array = utils.safe_cast_to_index(array)
         if dtype is None:
-            dtype = array.dtype
+            # if the index -'hides' its dtype such as
+            # PeriodIndex use an object dtype
+            if self.array._box_scalars:
```
Comment (shoyer) [inline xray/core/indexing.py]: It's probably a better idea to use `isinstance(self.array, pd.PeriodIndex)`? Relying on private API like `_box_scalars` has a tendency to get you burned.

Comment (maximilianr) [inline xray/core/indexing.py]: Yes; how do you balance that against the 'generalizable-ness' of this - will this then fail for `IntervalIndex` when that comes along, etc?

Comment (shoyer) [inline xray/core/indexing.py]: `IntervalIndex` will certainly not claim to have anything other than `dtype=object` :). I don't think we'll repeat the mistake of reporting bad dtypes again.

**Comment by shoyer** [comment]
Out of curiosity, what does the Dataset repr look like now when it contains a PeriodIndex?

It might be worth adding an integration test to explicitly verify that we can print such a Dataset without raising an error (no need to check the specific output, even, but just make sure it doesn't raise).

**Comment by maximilianr** [comment]
The repr hasn't changed - but the above example does work (matches the case that works)

File: xray/test/test_variable.py [hunk 2]
```diff
@@ -197,6 +197,12 @@ def test_pandas_data(self):
         v = self.cls(['x'], pd.Index([0, 1, 2]))
         self.assertEqual(v[0].values, v.values[0])
 
+    def test_pandas_period_index(self):
+        v = self.cls(['x'], pd.period_range(start='2000', periods=20, freq='B'))
+        self.assertEqual(v[0], pd.Period('2000', freq='B'))
+        # test repr works #GH645
+        repr(v)
```
Comment (shoyer) [inline xray/test/test_variable.py]: This actually hits a different codepath that never involves the logic in `core.formatting`. You'll need to put the variable in a Dataset if you want to see that.

File: xray/core/indexing.py (hunk 1 above)
Comment (shoyer) [inline xray/core/indexing.py]: A bigger concern is that xray will start crashing for every index type is the `_box_scalers` attribute goes away. At the very least, you need to use `getattr`.

**Comment by maximilianr** [comment]
@shoyer updates

**Comment by shoyer** [comment]
Thanks! There seems to be an issue with the conda matplotlib build on Windows which explains the appveyor failure.

**Pull request merged**